### PR TITLE
Add curriculum implementation

### LINKED ISSUE #24: Create tab layout with separate tab for curricula

### LINKED ISSUE #28: Create "grouped" curriculum view
New datafile

```yaml
- id: default
  groups:
      - title: "Software Basics"
         modules:
         - git
         - python
         - bash
     - title: "Software Development and Deployment"
        modules: 
        - git
        - advanced git
        - ci/cd (gitlab)
        - ci/cd (github)
```

For now, let's just create one curriculum "default" that is displayed on a separate tab of the main view

### PR DESCRIPTION
**Pull request by klieret**
Closes #28 
Closes #24

### CODE REVIEW

**Comment by ariostas** [comment]
This looks amazing! One minor thing I would suggest is to add a bit of spacing between curricula, because it looks a bit weird.

Screenshot 2023-12-19 at 9 42 08 AM

File: src/pages/index.js [hunk 1]
```diff
@@ -16,33 +19,68 @@ import Filter from "../components/filters";
...
+              return (
+                <div className="curriculumGroup-container" key={key}>
+                  <h3>{group.name}</h3>
+                  {group.description}
+                  <div className="tuts-container">
+                    {group.modules.map((module, key) => {
+                      var tut = tuts.find((x) => x.id == module.id);
+                      if (tut === undefined) {
+                        return "Undefined: " + module.id;
+                      }
```
[the remark this answers is not in the document]
Comment (ariostas) [reply src/pages/index.js:46]: I guess similar things are done in other places, so it doesn't make sense to only do it here.

**Review by ariostas — APPROVED** [review]

**Comment by ariostas** [comment]
One last thing, but probably not for this PR, is that the curricula filter is broken (or maybe it can be removed now).

Screenshot 2023-12-19 at 10 10 15 AM

**Comment by ariostas** [comment]
I made a minor change to add extra spacing between curricula groups. I'll merge this since I think it's great as it is and we can tweak things later if necessary. Thanks Kilian!

**Comment by klieret** [comment]
Thanks so much for the detailed review and fixing the spaces @ariostas !

File: src/pages/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (klieret) [reply src/pages/index.js:47]: Agree, this was done out of laziness. Let me open an issue for that

**Pull request merged**